components/dropdown/dropdown-content-mixin.js

```diff
@@ -0,0 +1,556 @@
+	 * `forceRender` may be used.
+	 */
+	async forceRender() {
+		if (!this.renderContent) {
+			this.renderContent = true;
+		}
+		await this.updateComplete;
+	}
+
+	/**
+	 * Toggles the opened/closed state of the dropdown.  If closed, it will open, and vice versa.
+	 * @param {Boolean} applyFocus Whether focus should be automatically moved to first focusable upon opening.
+	 */
+	toggleOpen(applyFocus) {
+		if (this.opened) {
+			this.close();
+		} else {
+			this.open(applyFocus);
+		}
+	}
+
+	/**
+	 * Private.
+	 */
+	scrollTo(scrollTop) {
+		const content = this.__content;
+		if (content) {
+			if (typeof scrollTop === 'number') {
+				content.scrollTop = scrollTop;
+			}
+			return content.scrollTop;
+		}
+	}
+
+	/**
+	 * Private.
+	 */
+	height() {
+		return this.__content && this.__content.offsetHeight;
+	}
+
+	__getContentContainer() {
+		return this.shadowRoot.querySelector('.d2l-dropdown-content-container');
+	}
+
+	__getPositionContainer() {
+		return this.shadowRoot.querySelector('.d2l-dropdown-content-position');
+	}
+
+	__getWidthContainer() {
+		return this.shadowRoot.querySelector('.d2l-dropdown-content-width');
+	}
+
+	__getOpener() {
+		const opener = findComposedAncestor(this, (elem) => {
+			if (elem.isDropdownOpener) {
+				return true;
+			}
+		});
+		return opener;
+	}
+
+	__onAutoCloseFocus() {
+
+		/* timeout needed to work around lack of support for relatedTarget */
+		setTimeout(() => {
+			if (!this.opened
+				|| this.noAutoClose
+				|| !document.activeElement
+				|| document.activeElement === this.__previousFocusableAncestor
+				|| document.activeElement === document.body) {
+				return;
+			}
+
+			const activeElement = getComposedActiveElement();
+
+			if (isComposedAncestor(this, activeElement)
+				|| isComposedAncestor(this.__getOpener(), activeElement)) {
+				return;
+			}
+
+			this.opened = false;
+		}, 0);
+	}
+
+	__onAutoCloseClick(e) {
+		if (!this.opened || this.noAutoClose) {
+			return;
+		}
+		const rootTarget = e.composedPath()[0];
+		const content = this.__getContentContainer();
+		if (isComposedAncestor(content, rootTarget)) {
+			return;
+		}
+		const opener = this.__getOpener();
+		if (isComposedAncestor(opener.getOpenerElement(), rootTarget)) {
+			return;
+		}
+
+		this.opened = false;
+	}
+
+	__onClose(e) {
+
+		if (e.target !== this || !document.activeElement) {
+			return;
+		}
+
+		const activeElement = getComposedActiveElement();
+
+		if (!isComposedAncestor(this, activeElement)) {
+			return;
+		}
+
+		const opener = this.__getOpener();
+		opener.getOpenerElement().focus();
+
+	}
+
+	__onResize() {
+		if (!this.opened) {
+			return;
+		}
+		this.__position();
+	}
+
+	async __openedChanged(newValue) {
+
+		this.__previousFocusableAncestor =
+			newValue === true
+				? getPreviousFocusableAncestor(this, false, false)
+				: null;
+
+		this.__isRTL = (getComputedStyle(this).direction === 'rtl');

```

Comment:
I think the `__isRTL` should no longer be needed.  We'll need to use the [RtlMixin](https://github.com/BrightspaceUI/core/blob/master/mixins/rtl-mixin.js) to properly handle the styles, and it will sprout `dir` on the host.  So we can avoid the `getComputedStyle` call (which can sometimes be costly) by using `this.getAttribute('dir')`. Referencing the property may also be an option.